interfaces/builtin/cifs_mount_control.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const cifsMountControlSummary = `allows mounting and unmounting CIFS filesystems`
+
+const cifsMountControlBaseDeclarationSlots = `
+  cifs-mount-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const cifsMountControlConnectedPlugSecComp = `
+# Description: Allow mount and umount syscall access.
+
+mount
+umount
+umount2
+`
+
+const cifsMountControlConnectedPlugAppArmor = `
+# Description: Allow mounting and unmounting CIFS filesystems.
+
+# Required for mounts and unmounts
+capability sys_admin,
+
+# Allow mounts to our snap-specific writable directories
+mount fstype=cifs ** -> @{HOMEDIRS}/*/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
+mount fstype=cifs ** -> @{HOMEDIRS}/*/snap/@{SNAP_NAME}/common/{,**},
+mount fstype=cifs ** -> /var/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
+mount fstype=cifs ** -> /var/snap/@{SNAP_NAME}/common/{,**},
+
+umount fstype=cifs @{HOMEDIRS}/*/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
+umount fstype=cifs @{HOMEDIRS}/*/snap/@{SNAP_NAME}/common/{,**},
+umount fstype=cifs /var/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**},
+umount fstype=cifs /var/snap/@{SNAP_NAME}/common/{,**},
+
+# Don't log violations for this file, see discussion here:
+#  - https://github.com/snapcore/snapd/pull/5340#issuecomment-398071797
+#  - https://forum.snapcraft.io/t/namespace-awareness-of-run-mount-utab-and-libmount/5987
+deny /run/mount/utab rw,
```

Comment:
I don't care for the explicit denial here since deny rules take precedent over allow rules. Ie, this means if another interface came along that allowed the access, this denial would still deny it. We either need to not worry about the logged denial or extend the comment for this. I prefer the former as a reminder to do something about https://forum.snapcraft.io/t/namespace-awareness-of-run-mount-utab-and-libmount/5987/3.

Reply:
removed the explicit denial and updated the comment